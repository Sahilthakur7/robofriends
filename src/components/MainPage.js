import React, { useEffect } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import '../containers/App.css';

const MainPage = ({searchField, onSearchChange, isPending, error, robots, onRequestRobots}) => {

  useEffect(() => {
    onRequestRobots()
  }, []);

  const filteredRobots = robots.filter((robot) => {
    return robot.name.toLowerCase().includes(searchField.toLowerCase());
  });

  return isPending ? (
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <h1 className="f1">RoboFriends</h1>
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <CardList robots={filteredRobots} />
      </Scroll>
    </div>
  );
};

export default MainPage;