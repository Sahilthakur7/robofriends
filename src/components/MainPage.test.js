import React from 'react';
import { shallow }  from 'enzyme';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
	const mockProps = {
		onRequestRobot: jest.fn(),
		robots: ['sahil', 'KV', 'KV2'],
		searchField: '',
		isPending: false,
	};

	wrapper = shallow(<MainPage {...mockProps} />);
});

it('Expects to render the main page component', () => {
	expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', () => {
	console.log("wrapper----", wrapper);
	const searchBox = wrapper.find('search-box');
	console.log("searchBox", searchBox)
})