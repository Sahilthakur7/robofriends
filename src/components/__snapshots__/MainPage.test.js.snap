// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expects to render the main page component 1`] = `undefined`;
